media/js/firefox/firstrun/membership.js

```diff
@@ -0,0 +1,214 @@
+            'eAction': 'Button Click',
+            'eLabel': e.target.dataset.label
+        });
+    }
+
+    function onButtonMouseOver(e) {
+        $slideShow.cycle('pause'); //pause auto-running slideshow
+        $slideShow.cycle('goto', e.target.dataset.index); //go to specific path
+    }
+
+    function onButtonMouseOut() {
+        $slideShow.cycle('resume'); //resume auto-running slideshow
+    }
+
+    function onLaterLinkClick(e) {
+        e.preventDefault();
+        window.dataLayer.push({
+            'event': 'in-page-interaction',
+            'eAction': 'Link Click',
+            'eLabel': 'Remind Me Later'
+        });
+    }
+
+    function bindEvents() {
+        for (var i = 0; i < button.length; i++) {
+            button[i].addEventListener('mouseover', onButtonMouseOver, false);
+            button[i].addEventListener('mouseout', onButtonMouseOut, false);
+            button[i].addEventListener('click', onButtonClick, false);
+        }
+
+        laterLink.addEventListener('click', onLaterLinkClick, false);
+    }
+
+    function unbindEvents() {
+        for (var i = 0; i < button.length; i++) {
+            button[i].removeEventListener('mouseover', onButtonMouseOver, false);
+            button[i].removeEventListener('mouseout', onButtonMouseOut, false);
+            button[i].removeEventListener('click', onButtonClick, false);
+        }
+
+        laterLink.removeEventListener('click', onLaterLinkClick, false);
+    }
+
+    function animateElement(selector, keyframes, options) {
+        return new Promise(function(resolve, reject) {
+            var element = document.querySelector(selector);
+
+            if (element) {
+                var animation = element.animate(keyframes, options);
+                animation.onfinish = function() {
+                    resolve(true);
+                };
+            } else {
+                reject('animateElement: Element for selector "' + selector +  '" not found.');
+            }
+        });
+    }
+
+    function fadeOutStep1() {
+        return new Promise(function(resolve, reject) {
+            var keyframes = [
+                { opacity: 1 },
+                { opacity: 0 }
+            ];
+
+            var options = {
+                duration: 200,
+                fill: 'forwards',
+                easing: 'ease-out',
+            };
+
+            animateElement('.step-1', keyframes, options).then(function() {
+                step1.classList.add('hidden');
+                resolve(true);
+            }).catch(function(reason) {
+                reject(reason);
+            });
+        });
+    }
+
+    function fadeInStep2() {
+        return new Promise(function(resolve, reject) {
+            var keyframes = [
+                { opacity: 0 },
+                { opacity: 1 }
+            ];
+
+            var options = {
+                duration: 200,
+                fill: 'forwards',
+                easing: 'ease-in',
+            };
```

Comment:
I'm wondering, the `keyframes` and `options` here are almost identical to those in `fadeOutStep1`. I would suggest that you have a config object as follows:

```
var config = {
    keyframes: [
        { opacity: 0 },
        { opacity: 1 }
    ],
    options: {
        duration: 200,
        fill: 'forwards',
        easing: 'ease-out',
    }
}
```

Reply:
Well, they are similar but not quite. The keyframes are reversed, and the easing also changes. I could use a shared config object and then override things where needed, but I'm not sure this would make the code any easier to follow. Thoughts?

Reply:
True. I still like the idea of a shared `config` object, it’s also a few lines less code ;) I do not have strong feelings on this so, I would say, either way is fine.

Reply:
Yeah it does save a couple of lines of code, but one thing I'm trying to do these days is avoid writing functions that produce side-effects (i.e. altering the config object). This is relatively safe as there would be only two functions that consume it, but if this page we're more complicated it could lead to possible errors. I'm going to leave it be for now 👍